### ISSUE TITLE
Changing kafka.partitionLimitation seems to break consuming

### ISSUE THREAD

**Issue by AndreasBergmeier6176** [kedacore/keda#4328]
### Report

When we switch from specifying `partitionLimitation` (`= 1`) to setting it to empty (`""`) it seems like Keda no longer consumes messages.

### Expected Behavior

Keda should switch from partition assignment to consuming all partitions again.

### Actual Behavior

Keda scales to 0. We get a KEDAScaleTargetDeactivated event.

### Steps to Reproduce the Problem

1. Deploy `ScaledObject` with `partitionLimitation = 1`
2. Change `partitionLimitation` to `""`
3. Wait for Event on Deployment

### Logs from KEDA operator

The only log line for our deployment `foo` in the operator log is this:
```
"2023-03-07T08:57:32Z	INFO	Initializing Scaling logic according to ScaledObject Specification	{"controller": "scaledobject", "controllerGroup": "keda.sh", "controllerKind": "ScaledObject", "scaledObject": {"name":"foo","namespace":"bar"}, "namespace": "bar", "name": "foo", "reconcileID": "72abb643-49ec-4310-b34f-888bf7e489a7"}"
```

### KEDA Version

2.8.2

### Kubernetes Version

1.25

### Platform

Google Cloud

### Scaler Details

Kafka

### Anything else?

Going from `""` to `1` makes it work again though.

**Comment by JorTurFer** [comment]
Hi,
Could you confirm what KEDA version you are using? Could you also share KEDA operator logs?

**Comment by JorTurFer** [comment]
I need to know what KEDA version you are using please
Could you also share KEDA operator logs?

**Comment by AndreasBergmeier6176** [comment]
> I need to know what KEDA version you are using
We are using 2.8.2 with the `partitionLimitation` patch.
Sadly we cannot upgrade to 2.9 since you broke multi-cert support with 2.9 :(

> Could you also share KEDA operator logs?
Cleaning the full log will take me some time.
You could also tell me what you are looking for - might make it easier...

**Comment by AndreasBergmeier6176** [comment]
> What do you mean? We plan to release v2.10 soon and if something is broken and we know it, we can try to solve it for incoming release.

Previously we could add rolling (2+) CAs. With 2.9 this no longer works.

**Comment by JorTurFer** [comment]
> Previously we could add rolling (2+) CAs. With 2.9 this no longer works.

Could you be more explicit? How do you do it? It's the first time that I have heard about this. 
Could this new feature that will be released as part of v2.10 solve your CAs issues? 
PR: https://github.com/kedacore/keda/pull/4191
Docs: https://keda.sh/docs/2.10/operate/security/#register-your-own-ca-in-keda-operator-trusted-store

**Comment by JorTurFer** [comment]
But just to understand your use case, you are adding some custom CAs to KEDA in order to trust in them during HTTP requests, right?
How are you adding those CAs to KEDA?

**Comment by AndreasBergmeier6176** [comment]
Tried to create a new Bug for the upgrade problem: https://github.com/kedacore/keda/issues/4330

**Comment by JorTurFer** [comment]
Yes, all the custom CAs will be added to any place where an HTTP client is used. In this case [kafka client uses it](https://github.com/kedacore/keda/pull/4191/files#diff-03cf62956adcec86c0717723ab146472f0979221b0e5942a6e5460fa6d5e4f08R321), so any custom CA will be added to it

**Comment by JorTurFer** [comment]
Is this solved by #4333?

**Comment by AndreasBergmeier6176** [comment]
> Is this solved by #4333?

Will try to create a setup with `main` tomorrow. Probably not, though.
